### ISSUE TITLE
sinon.test() does not pass on errors to callbacks

### ISSUE THREAD

**Issue by fatso83** [cjohansen/Sinon.JS#818]
`sinon.test(fn)` is supposedly supposed to do the same as `fn` (except for sandboxing), but this is not true when it comes to callbacks, as the following tests are meant to illustrate.

```javascript
it('works', function () {

    var normalFn= (function (callback) {
        callback(new Error('myerror'));
    });

    function errorChecker (err) {
        expect(err).to.not.be(undefined);
        errorChecker.called = true
    }

    normalFn(errorChecker)
    expect(errorChecker.called).to.be(true);
});

it('shows sinon.test bug - as this fails', function () {

    var sandboxedFn = sinon.test(function (callback) {
        callback(new Error('myerror'));
    });

    // is never called
    function errorChecker (err) {
        expect(err).to.not.be(undefined);
        errorChecker.called = true
    }

    sandboxedFn(errorChecker)
    expect(errorChecker.called).to.be(true);
});
```

I think the bug is caused by [some code in `lib/sinon/test.js`](https://github.com/cjohansen/Sinon.JS/blob/bff6358be696d3c662557cef96345468007762eb/lib/sinon/test.js#L36), but as I am not quite sure how it works, I haven't fixed it yet.
```javascript
if (typeof oldDone === "function") {
    args[args.length - 1] = function sinonDone(res) {
        if (res) {
            sandbox.restore();
            throw exception;
        } else {
            sandbox.verifyAndRestore();
        }
        oldDone(res);
    };
}```

**Comment by mroderick** [comment]
https://github.com/cjohansen/Sinon.JS/blob/master/CONTRIBUTING.md#reporting-an-issue

**Comment by mroderick** [comment]
Whoops, wrong button

**Comment by fatso83** [comment]
Yeah, sorry about submitting before fleshing out the details; was caused by a "Tab,Enter" user bug ... Detailing it now.

**Comment by fatso83** [comment]
This behavior is verified in Chrome 44 and PhantomJS 1.9.2, btw (through the Buster tests). Anything else that prevents the PR from being merged (except for time)?

**Comment by mroderick** [comment]
Fixed via #820
